SourceDistributions/Developer-Sandboxes/tsWxGTUI_PyVx/Python-3x-3.6.0/tsWxGTUI_Py3x-3.6.0/ManPages/tsManPagesLibGUI/tsWxPasswordEntryDialog.man

Help on module tsWxPasswordEntryDialog:

NAME
    tsWxPasswordEntryDialog

FILE
    ./tsLibGUI/tsWxPkg/src/tsWxPasswordEntryDialog.py

DESCRIPTION
    tsWxPasswordEntryDialog.py - Class represents a dialog that
    requests a one-line password string from the user.

CLASSES
    tsWxTextEntryDialog.TextEntryDialog(tsWxDialog.Dialog)
        PasswordEntryDialog
    
    class PasswordEntryDialog(tsWxTextEntryDialog.TextEntryDialog)
     |  Class represents a dialog that requests a one-line text string
     |  from the user.
     |  
     |  Method resolution order:
     |      PasswordEntryDialog
     |      tsWxTextEntryDialog.TextEntryDialog
     |      tsWxDialog.Dialog
     |      tsWxTopLevelWindow.TopLevelWindow
     |      tsWxWindow.Window
     |      tsWxEvtHandler.EvtHandler
     |      tsWxObject.Object
     |      __builtin__.object
     |  
     |  Methods defined here:
     |  
     |  __init__(self, parent, message, caption='', value='', style=2048, size=(-1, -1), pos=(-1, -1))
     |      Use ShowModal() to show the dialog.
     |      
     |      Parameters:
     |      
     |      parent  Parent window.
     |      
     |      message Message to show on the dialog.
     |      
     |      caption The caption of the dialog.
     |      
     |      value   The default value, which may be the empty string.
     |      
     |      style   A dialog style, specifying the buttons (wxOK, wxCANCEL)
     |              and an optional wxCENTRE style. Additionally, wxTextCtrl
     |              styles (such as wxTE_PASSWORD or wxTE_MULTILINE) may be
     |              specified here.
     |      
     |      pos     Dialog position.
     |  
     |  ----------------------------------------------------------------------
     |  Methods inherited from tsWxTextEntryDialog.TextEntryDialog:
     |  
     |  GetValue(self)
     |      Returns the text that the user has entered if the user has pressed
     |      OK, or the original value if the user has pressed Cancel.
     |  
     |  SetValue(self, value)
     |      Sets the default text value.
     |  
     |  ShowModal(self)
     |      Shows the dialog, returning wx.ID_OK if the user pressed OK,
     |      and wx.ID_CANCEL otherwise.
     |  
     |  tsDialogFeatureLayout(self)
     |  
     |  tsOnCancelButtonClick(self, evt)
     |  
     |  tsOnOkButtonClick(self, evt)
     |  
     |  tsTextEntryDialogLayout(self, parent, pos, size, style, name)
     |      Calculate position and size of dialog based upon arguments.
     |  
     |  ----------------------------------------------------------------------
     |  Methods inherited from tsWxDialog.Dialog:
     |  
     |  Create(self, parent, id=-1, title='', pos=(-1, -1), size=(-1, -1), style=570433088, name='dialog', pixels=True)
     |      Create the GUI part of the Window for 2-phase creation mode.
     |  
     |  CreateButtonSizer(self, flags, *ignored)
     |  
     |  CreateSeparatedButtonSizer(self, flags)
     |  
     |  CreateStdDialogButtonSizer(self, flags)
     |  
     |  CreateTextSizer(self, message)
     |  
     |  EndModal(self, retCode)
     |  
     |  GetAffirmativeId(self)
     |  
     |  GetEscapeId(self)
     |  
     |  GetReturnCode(self)
     |  
     |  IsModal(self)
     |  
     |  SetAffirmativeId(self, affirmativeId)
     |  
     |  SetEscapeId(self, escapeId)
     |  
     |  SetReturnCode(self, returnCode)
     |  
     |  tsBuildOneButton(self, parent, label=None, event=None, handler=None, pos=(0, 0), name='button')
     |      Create the specified Frame Button located on the Frame Title
     |      Line.
     |      
     |      Based on "wxPython In Action" by Noel Rappin and Robin Dunn,
     |      Manning Publications Co. (See downloaded source code located in
     |      Chapter-05, goodExampple.py.)
     |  
     |  tsButtonData(self)
     |      Define the label and handler for the Iconize, Maximize and Close
     |      Buttons on the Frame Title Line.
     |      
     |      Based on "wxPython In Action" by Noel Rappin and Robin Dunn,
     |      Manning Publications Co. (See downloaded source code located in
     |      Chapter-05, goodExampple.py.)
     |  
     |  tsCreateButtonBar(self, panel, xPos=0, yPos=0)
     |      Create the collection of Dialog Buttons located on the Dialog Title
     |      Line.
     |      
     |      Based on "wxPython In Action" by Noel Rappin and Robin Dunn,
     |      Manning Publications Co. (See downloaded source code located in
     |      Chapter-05, goodExampple.py.)
     |  
     |  tsDialogWindowLayout(self, parent, pos, size, style, name)
     |      Calculate position and size of dialog based upon arguments.
     |  
     |  tsGetClientArea(self, pixels=True)
     |      Returns the bounding rectangle the client area of the Dialog.
     |  
     |  tsTrapIfTooSmall(self, name, myRect)
     |  
     |  tsUpdate(self)
     |      Draw the actual features of the Dialog.
     |  
     |  ----------------------------------------------------------------------
     |  Static methods inherited from tsWxDialog.Dialog:
     |  
     |  GetClassDefaultAttributes(variant=0)
     |      Get the default attributes for this class. This is useful if you want
     |      to use the same font or colour in your own control as in a standard
     |      control -- which is a much better idea than hard coding specific
     |      colours or fonts which might look completely out of place on the
     |      users system, especially if it uses themes.
     |      
     |      The variant parameter is only relevant under Mac currently and is
     |      ignore under other platforms. Under Mac, it will change the size of
     |      the returned font. See wx.Window.SetWindowVariant for more about this.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from tsWxDialog.Dialog:
     |  
     |  AffirmativeId
     |  
     |  ClientArea
     |      Returns the bounding rectangle the client area of the Dialog.
     |  
     |  EscapeId
     |  
     |  ReturnCode
     |  
     |  ----------------------------------------------------------------------
     |  Methods inherited from tsWxTopLevelWindow.TopLevelWindow:
     |  
     |  EnableCloseButton(self, enable)
     |      Return True if close button enabled state is being changed.
     |      Return False if close button enabled state is NOT being changed.
     |  
     |  GetClosed(self)
     |      Return True if top level window has been closed.
     |      Return False if top level window has NOT been closed.
     |  
     |  GetDefaultItem(self)
     |      Get the default child of this parent, i.e. the first one
     |  
     |  GetHelpAsked(self)
     |      Return True if top level window has been asked for Help.
     |      Return False if top level window has NOT been asked for Help.
     |  
     |  GetIcon(self)
     |      TBD
     |  
     |  GetIconized(self)
     |      Return True if top level window has been iconized.
     |      Return False if top level window has NOT been iconized.
     |  
     |  GetMaximized(self)
     |      Return True if top level window has been maximuzed.
     |      Return False if top level window has NOT been maximized.
     |  
     |  GetRestoreDown(self)
     |      Return True if top level window has been restoredown.
     |      Return False if top level window has NOT been restoredown.
     |  
     |  GetTitle(self)
     |      TBD - Return title or None as appropriate.
     |  
     |  GetTmpDefaultItem(self)
     |      Return the temporary default item, which can be None.
     |  
     |  Iconize(self, iconize)
     |  
     |  IsActive(self)
     |  
     |  IsAlwaysMaximized(self)
     |  
     |  IsClosed(self)
     |  
     |  IsFullScreen(self)
     |  
     |  IsHelpAsked(self)
     |  
     |  IsIconized(self)
     |  
     |  IsMaximized(self)
     |  
     |  IsRestoreDown(self)
     |  
     |  MacGetMetalAppearance(self)
     |  
     |  MacGetUnifiedAppearance(self)
     |  
     |  MacSetMetalAppearance(self, on)
     |  
     |  Maximize(self, maximize)
     |  
     |  RequestUserAttention(self, flags)
     |  
     |  SetClosed(self, closed)
     |  
     |  SetDefaultItem(self, child)
     |      Set this child as default, return the old default.
     |  
     |  SetHelpAsked(self, helpAsked)
     |  
     |  SetIcon(self, icon)
     |  
     |  SetIconized(self, iconized)
     |  
     |  SetIcons(self, icons)
     |  
     |  SetMaximized(self, maximized)
     |  
     |  SetRestoreDown(self, restoredown)
     |  
     |  SetShape(self, region)
     |  
     |  SetTitle(self, title)
     |  
     |  SetTmpDefaultItem(self, win)
     |      Set this child as temporary default
     |  
     |  ShowFullScreen(self, show, style)
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from tsWxTopLevelWindow.TopLevelWindow:
     |  
     |  Closed
     |      Return True if top level window has been closed.
     |      Return False if top level window has NOT been closed.
     |  
     |  DefaultItem
     |      Get the default child of this parent, i.e. the first one
     |  
     |  HelpAsked
     |      Return True if top level window has been asked for Help.
     |      Return False if top level window has NOT been asked for Help.
     |  
     |  Icon
     |      TBD
     |  
     |  Iconized
     |      Return True if top level window has been iconized.
     |      Return False if top level window has NOT been iconized.
     |  
     |  Maximized
     |      Return True if top level window has been maximuzed.
     |      Return False if top level window has NOT been maximized.
     |  
     |  RestoreDown
     |      Return True if top level window has been restoredown.
     |      Return False if top level window has NOT been restoredown.
     |  
     |  Title
     |      TBD - Return title or None as appropriate.
     |  
     |  TmpDefaultItem
     |      Return the temporary default item, which can be None.
     |  
     |  ----------------------------------------------------------------------
     |  Methods inherited from tsWxWindow.Window:
     |  
     |  AcceptsFocus(self)
     |      This method may be overridden in the derived classes to return false
     |      to indicate that this control does not accept input at all
     |  
     |  AcceptsFocusFromKeyboard(self)
     |      This method may be overridden in the derived classes to return false
     |      to indicate that while this control can, in principle, have focus if
     |      the user clicks it with the mouse, it should not be included in the
     |      TAB traversal chain when using the keyboard.
     |  
     |  AcceptsFocusRecursively(self)
     |      Overridden to indicate whether this window or one of its children
     |      accepts focus.
     |  
     |  AddChild(self, child)
     |      Adds a child window. This is called automatically by window creation
     |      functions so should not be required by the application programmer.
     |  
     |  AddPendingEvent(self, event)
     |      See ProcessEvent() for more info about why you should not use
     |      this function and the reason for making this function protected
     |      in wxWindow.
     |      
     |      Reimplemented from wxEvtHandler.
     |  
     |  AdjustForLayoutDirection(self, x, width, widthTotal)
     |      Mirror coordinates for RTL layout if this window uses it and
     |      if the mirroring is not done automatically like Win32.
     |  
     |  AdjustForParentClientOriginXY(self, x, y, sizeFlags)
     |  
     |  AlwaysShowScrollbars(self)
     |      Call this function to force one or both scrollbars to be always
     |      shown, even if the window is big enough to show its entire contents
     |      without scrolling.
     |  
     |  AssociateHandle(self, handle)
     |      Associate the window with a new native handle.
     |  
     |  CacheBestSize(self, size)
     |      Sets the cached best size value.
     |  
     |  CanScroll(self, orient)
     |      Returns True if this window can have a scroll bar in this
     |      orientation.
     |  
     |  CanSetTransparent(self)
     |      Returns true if the system supports transparent windows and
     |      calling SetTransparent() may succeed.
     |      
     |      If this function returns false, transparent windows are definitely
     |      not supported by the current system.
     |      
     |      Reimplemented in wxTopLevelWindow.
     |      
     |      NOTE: On X-windows systems the X server must have the composite
     |      extension loaded, and there must be a composite manager program
     |      (such as xcompmgr) running.
     |  
     |  CaptureMouse(self)
     |      Directs all mouse input to this window.
     |      
     |      Call ReleaseMouse() to release the capture.
     |      
     |      Note that wxWindows maintains the stack of windows having captured
     |      the mouse and when the mouse is released the capture returns to the
     |      window which had had captured it previously and it is only really
     |      released if there were no previous window. In particular, this means
     |      that you must release the mouse as many times as you capture it,
     |      unless the window receives the wx.MouseCaptureLostEvent event.
     |      
     |      Any application which captures the mouse in the beginning of some
     |      operation must handle wx.MouseCaptureLostEvent and cancel this
     |      operation when it receives the event. The event handler must not
     |      recapture mouse.
     |  
     |  Center(self, direction=12)
     |      Centers the window. The parameter specifies the direction for
     |      cetering, and may be wx.HORIZONTAL, wx.VERTICAL or wx.BOTH. It may
     |      also include wx.CENTER_ON_SCREEN flag if you want to center the
     |      window on the entire screen and not on its parent window. If it is
     |      a top-level window and has no parent then it will always be centered
     |      relative to the screen.
     |  
     |  CenterOnParent(self, dir=12)
     |      Center with respect to the the parent window.
     |  
     |  CenterOnScreen(self, direction=12)
     |      Center the window on screen.
     |  
     |  Centre(self, direction=12)
     |      Centers the window. The parameter specifies the direction for
     |      cetering, and may be wx.HORIZONTAL, wx.VERTICAL or wx.BOTH. It may
     |      also include wx.CENTER_ON_SCREEN flag if you want to center the
     |      window on the entire screen and not on its parent window. If it is
     |      a top-level window and has no parent then it will always be centered
     |      relative to the screen.
     |  
     |  CentreOnParent(self, dir=12)
     |      Center with respect to the the parent window.
     |  
     |  CentreOnScreen(self, direction=12)
     |      Center the window on screen.
     |  
     |  ClearBackground(self)
     |      Clears the window by filling it with the current background colour.
     |      
     |      Does not cause an erase background event to be generated.
     |      
     |      Notice that this uses wxClientDC to draw on the window and the
     |      results of doing it while also drawing on wxPaintDC for this
     |      window are undefined. Hence this method should not be used
     |      from EVT_PAINT handlers, just use wxDC::Clear() on the wxPaintDC
     |      you already use there instead.
     |  
     |  ClientToScreen(self, pt)
     |      Converts to screen coordinates from coordinates relative to this
     |      window.
     |  
     |  ClientToScreenXY(self, x, y)
     |      Converts to screen coordinates from coordinates relative to this
     |      window.
     |  
     |  ClientToWindowSize(self, size)
     |      Converts client area size size to corresponding window size.
     |  
     |  Close(self, force=True)
     |      This function simply generates a wxCloseEvent whose handler usually
     |      tries to close the window.
     |      
     |      It does not close the window itself, however.
     |      
     |      Parameters:
     |      force   false if the windows close handler should be able to veto
     |      the destruction of this window, true if it cannot.
     |      
     |      Remarks:
     |      Close calls the close handler for the window, providing an
     |      opportunity for the window to choose whether to destroy the window.
     |      Usually it is only used with the top level windows (wxFrame and
     |      wxDialog classes) as the others are not supposed to have any special
     |      OnClose() logic. The close handler should check whether the window
     |      is being deleted forcibly, using wxCloseEvent::CanVeto, in which
     |      case it should destroy the window using wxWindow::Destroy. Note
     |      that calling Close does not guarantee that the window will be
     |      destroyed; but it provides a way to simulate a manual close of
     |      a window, which may or may not be implemented by destroying the
     |      window. The default implementation of wxDialog::OnCloseWindow
     |      does not necessarily delete the dialog, since it will simply
     |      simulate an wxID_CANCEL event which is handled by the appropriate
     |      button event handler and may do anything at all. To guarantee
     |      that the window will be destroyed, call wxWindow::Destroy instead.
     |  
     |  ConvertDialogPixels(self, pt)
     |      Converts a point or size from dialog units to pixels.
     |      
     |      For the x dimension, the dialog units are multiplied by the
     |      fixed point character width. For the y dimension, the dialog
     |      units are multiplied by the fixed point character height.
     |      
     |      Remarks:
     |      Dialog units are used for maintaining a dialogs proportions
     |      even if the font changes. You can also use these functions
     |      programmatically. A convenience macro is defined:
     |      define wxDLG_UNIT(parent, pt) parent->ConvertDialogToPixels(pt)
     |  
     |  ConvertDialogPointToPixels(self, pt)
     |      Converts a point or size from dialog units to pixels.
     |  
     |  ConvertDialogSizeToPixels(self, sz)
     |      Converts a point or size from dialog units to pixels.
     |  
     |  ConvertPixelPointToDialog(self, pt)
     |      Converts a point or size from pixel units to dialog units.
     |  
     |  ConvertPixelSizeToDialog(self, sz)
     |      Converts a point or size from pixel units to dialog units.
     |  
     |  ConvertPixelsToDialog(self, pt)
     |      Converts a point or size from pixels to dialog units.
     |      
     |      For the x dimension, the pixels are divided by the fixed point
     |      character width. For the y dimension, the pixels are divided
     |      by the fixed point character height.
     |      
     |      Remarks:
     |      Dialog units are used for maintaining a dialogs proportions even
     |      if the font changes.
     |  
     |  DLG_PNT(self, pt)
     |      Converts a point or size from dialog units to pixels. Dialog units
     |      are used for maintaining a dialog proportions even if the font
     |      changes. For the x dimension, the dialog units are multiplied by the
     |      average character width and then divided by 4. For the y dimension,
     |      the dialog units are multiplied by the average character height and
     |      then divided by 8.
     |  
     |  DLG_SZE(self, sz)
     |      Converts a point or size from dialog units to pixels. Dialog units
     |      are used for maintaining a dialog proportions even if the font
     |      changes. For the x dimension, the dialog units are multiplied by the
     |      average character width and then divided by 4. For the y dimension,
     |      the dialog units are multiplied by the average character height and
     |      then divided by 8.
     |  
     |  Destroy(self)
     |      Destroys the window safely.
     |      
     |      Use this function instead of the delete operator, since different
     |      window classes can be destroyed differently. Frames and dialogs
     |      are not destroyed immediately when this function is called -- they
     |      are added to a list of windows to be deleted on idle time, when
     |      all the windows events have been processed. This prevents problems
     |      with events being sent to non-existent windows.
     |      
     |      Returns:
     |      true if the window has either been successfully deleted, or it has
     |      been added to the list of windows pending real deletion.
     |  
     |  DestroyChildren(self)
     |      Destroys all children of a window.
     |  
     |  Disable(self)
     |      Disables the window for user input, same as Enable(False).
     |  
     |  DissociateHandle(self)
     |      Dissociate the current native handle from the window.
     |  
     |  DoCaptureMouse(self)
     |  
     |  DoCentre(self, dir)
     |      Centres the window.
     |      
     |      Parameters:
     |      direction  Specifies the direction for the centring. May be
     |      wxHORIZONTAL, wxVERTICAL or wxBOTH. It may also include the
     |      wxCENTRE_ON_SCREEN flag.

Documents/NEWS.txt
#-----------------------------------------------------------
#"Time-stamp: <07/08/2015  4:48:41 AM rsg>"
#-----------------------------------------------------------

========================= NEWS.txt =========================

   +----+----+  TeamSTARS "tsWxGTUI_PyVx" Toolkit
   | ts | Wx |      with Python 2x & Python 3x based
   +----+----+         Command Line Interface (CLI)
   | G T U I |      and "Curses"-based "wxPython"-style, 
   +---------+         Graphical-Text User Interface (GUI)
 
   Get that cross-platform, pixel-mode "wxPython" feeling
   on character-mode 8-/16-color (xterm-family) & non-color
   (vt100-family) terminals and terminal emulators.

   You can find this and other plain-text files in the
   Toolkit subdirectory named:

       "./<Toolkit Recipient's Repository>/Documents".

       <Your Working Repository>
       (e.g. "tsWxGTUI_PyVx_Repository") 
         |
         |
         +-- ["Documents"]

=========================== News ===========================

The pre-alpha release of the TeamSTARS "tsWxGTUI_PyVx-0.0.0"
Toolkit is now available. It includes the following two
versions abd their associated variants (Python Distribution
Utilities ("Distutils") installable "Site-Packages" and non-
installable "Developer-Sandboxes" for enhancement and bug
fix development and experimentation without interference
with the already installed Site-Packages):

1. The TeamSTARS "tsWxGTUI_Py2x-0.0.0" Toolkit is released
   for users of the popular, high-level, cross-platform,
   second generation Python 2.x programming language which
   is widely available.

   A wiki posting by the Python Software Foundation announc-
   ed that Python 3.0 was released in 2008. The final 2.x
   version 2.7 release came out in mid-2010, with a state-
   ment of extended support for this end-of-life release.
   The 2.x branch will see no new major releases after that.

   The source code and documentation is available for down-
   load at:

      http://pypi.python.org/pypi.

   It supports development and operation of:

   a. Command Line User Interface (CLI) application
      programs implemented in Python 2.0.0-2.7.9.

   b. Graphical-style User Interface (GUI) application
      programs implemented in Python 2.6.4-2.7.9.

2. The TeamSTARS "tsWxGTUI_Py3x-0.0.0" Toolkit is released
   for users of the popular, high-level, cross-platform,
   third generation Python 3.x programming language which
   is growing in availability.

   A wiki posting by the Python Software Foundation announc-
   ed that Python 3.0 was released in 2008. 3.x is under
   active development and has already seen over five years
   of stable releases, including version 3.3 in 2012 and
   3.4 in 2014. This means that all recent standard library
   improvements, for example, are only available by default
   in Python 3.x.

   Guido van Rossum (the original creator of the Python
   language) decided to clean up Python 2.x properly,
   with less regard for backwards compatibility than is
   the case for new releases in the 2.x range. The most
   drastic improvement is the better Unicode support
   (with all text strings being Unicode by default) as
   well as saner bytes/Unicode separation.

   Besides, several aspects of the core language (such as
   print and exec being statements, integers using floor
   division) have been adjusted to be easier for newcomers
   to learn and to be more consistent with the rest of the
   language, and old cruft has been removed (for example,
   all classes are now new-style, "range()" returns a memory
   efficient iterable, not a list as in 2.x).

   The source code and documentation is available for down-
   load at:

      http://pypi.python.org/pypi.

   It supports development and operation of:

   a. Command Line User Interface (CLI) application
      programs implemented in Python 3.0.0-3.4.2.

   b. Graphical-style User Interface (GUI) application
      programs implemented in Python 3.1.5-3.4.2.

======================= End-Of-File ========================


SourceDistributions/Developer-Sandboxes/tsWxGTUI_PyVx/Python-3x/tsWxGTUI_Py3x/tsLibGUI/tsWxPkg/src/tsWxPasswordEntryDialog.py
#! /usr/bin/env python
# "Time-stamp: <10/25/2013  6:41:34 AM rsg>"
'''
tsWxPasswordEntryDialog.py - Class represents a dialog that
requests a one-line password string from the user.
'''
#################################################################
#
# File: tsWxPasswordEntryDialog.py
#
# Purpose:
#
#    Class represents a dialog that requests a one-line password
#    string from the user.
#
# Usage (example):
#
#    # Import
#
#    from tsWxPasswordEntryDialog import PasswordEntryDialog
#
# Requirements:
#
#    wxPython 2.8.9.2 (New wxPyDocs)/wxWidgets 2.9.2 (Ref. Man.).
#
# Capabilities:
#
#    wxPython 2.8.9.2 (New wxPyDocs)/wxWidgets 2.9.2 (Ref. Man.).
#
# Limitations:
#
#    wxPython 2.8.9.2 (New wxPyDocs)/wxWidgets 2.9.2 (Ref. Man.).
#
# Notes:
#
#    None
#
# Methods:
#
#    wxPython 2.8.9.2 (New wxPyDocs)/wxWidgets 2.9.2 (Ref. Man.).
#
# Classes:
#
#    wxPython 2.8.9.2 (New wxPyDocs)/wxWidgets 2.9.2 (Ref. Man.).
#
# Modifications:
#
#   2012/08/12 rsg Changed tsWxTextEntryDialog and tsWxTextEditBox.
#                  Resolved inability to turn-off the echoing of
#                  input characters and the inability to stop
#                  displaying dialog with a visible password.
#                  Resolution applicable only for event driven mode.
#
# ToDo:
#
#   2012/07/12 rsg Investigate and resolve inability to turn-off
#                  the echoing of input characters and the
#                  inability to stop displaying dialog with
#                  a visible password.
#
#################################################################

__title__     = 'tsWxPasswordEntryDialog'
__version__   = '1.1.0'
__date__      = '04/01/2013'
__authors__   = 'Richard S. Gordon'
__copyright__ = 'Copyright (c) 2007-2013 ' + \
                '%s.\n\t\tAll rights reserved.' % __authors__
__license__   = 'GNU General Public License, ' + \
                'Version 3, 29 June 2007'
__credits__   = '\n\n  Credits: ' + \
                '\n\n\t  tsLibGUI Import & Application Launch Features: ' + \
                '\n\t  Copyright (c) 2007-2009 Frederick A. Kier.' + \
                '\n\t\t\tAll rights reserved.' + \
                '\n\n\t  Python Curses Module API & ' + \
                'Run Time Library Features:' + \
                '\n\t  Copyright (c) 2001-2013 ' +\
                'Python Software Foundation.' + \
                '\n\t\t\tAll rights reserved.' + \
                '\n\t  PSF License Agreement for Python 2.7.3 & 3.3.0' + \
                '\n\n\t  wxWidgets (formerly wxWindows) & ' + \
                'wxPython API Features:' + \
                '\n\t  Copyright (c) 1992-2008 Julian Smart, ' + \
                'Robert Roebling,' + \
                '\n\t\t\tVadim Zeitlin and other members of the ' + \
                '\n\t\t\twxWidgets team.' + \
                '\n\t\t\tAll rights reserved.' + \
                '\n\t  wxWindows Library License' + \
                '\n\n\t  nCurses API & Run Time Library Features:' + \
                '\n\t  Copyright (c) 1998-2011 ' + \
                'Free Software Foundation, Inc.' + \
                '\n\t\t\tAll rights reserved.' + \
                '\n\t  GNU General Public License, ' + \
                'Version 3, 29 June 2007'

__line1__ = '%s, v%s (build %s)' % (__title__, __version__, __date__)
__line2__ = 'Author(s): %s' % __authors__
__line3__ = '%s' % __copyright__

if len( __credits__) == 0:
    __line4__ = '%s' % __license__
else:
    __line4__ = '%s%s' % (__license__, __credits__)

__header__ = '\n\n%s\n\n  %s\n  %s\n  %s\n' % (__line1__,
                                               __line2__,
                                               __line3__,
                                               __line4__)

mainTitleVersionDate = __line1__

from tsWxDoubleLinkedList import DoubleLinkedList as wxDoubleLinkedList
from tsWxTextEntryDialog import TextEntryDialog
import tsWxGlobals as wx

#---------------------------------------------------------------------------

##DEBUG = True
DEBUG = wx.Debug_GUI_Launch | \
        wx.Debug_GUI_Progress | \
        wx.Debug_GUI_Termination | \
        wx.Debug_GUI_Exceptions

##VERBOSE = True
VERBOSE = wx.Debug_GUI_Configuration

#---------------------------------------------------------------------------

class PasswordEntryDialog(TextEntryDialog):
    '''
    Class represents a dialog that requests a one-line text string
    from the user.
    '''
    def __init__(self,
                 parent,                 # Should be None unless modal mode
                 message,                # Alias for user prompt output
                 caption=wx.EmptyString, # Alias for title
                 value=wx.EmptyString,   # Option for default user input
                 style=wx.TE_PASSWORD,   # Options for ClientRect
                 size=wx.DefaultSize,    # Extension for application use
                 pos=wx.DefaultPosition  # Option for application use
                 ):
        '''
        Use ShowModal() to show the dialog.

        Parameters:

        parent  Parent window.

        message Message to show on the dialog.

        caption The caption of the dialog.

        value   The default value, which may be the empty string.

        style   A dialog style, specifying the buttons (wxOK, wxCANCEL)
                and an optional wxCENTRE style. Additionally, wxTextCtrl
                styles (such as wxTE_PASSWORD or wxTE_MULTILINE) may be
                specified here.

        pos     Dialog position.
        '''
        theClass = 'PasswordEntryDialog'

        # Capture initial caller parametsrs before Dialog makes changes
        self.backup_parent = parent
        self.backup_message = message
        self.backup_caption = caption
        self.backup_value = value
        self.backup_style = style
        self.backup_size = size
        self.backup_pos = pos

        name = wx.PasswordEntryDialogNameStr
        title = caption
        self.backup_name = name
        self.backup_size = size
        self.backup_title = title

        # Capture initial caller parametsrs before they are changed
        self.caller_caption = caption
        self.caller_message = message
        self.caller_name = name
        self.caller_parent = parent
        self.caller_pos = pos
        self.caller_size = size
        self.caller_style = style
        self.caller_title = title
        self.caller_value = value

        wx.RegisterFirstCallerClassName(self, theClass)

        TextEntryDialog.__init__(
            self,
            parent,          # Should be None unless modal mode
            message,         # Alias for user prompt output
            caption=caption, # Alias for title
            value=value,     # Option for default user input
            style=style,     # Options for ClientRect
            size=size,       # Extension for application use
            pos=pos          # Option for application use
            )

        # Capture parameters.
        self.ts_Caption = self.backup_caption
        self.ts_Message = self.backup_message
        self.ts_Parent = self.backup_parent    # Should be None unless modal mode
        self.ts_Pos = self.backup_pos
        self.ts_Size = self.backup_size
        self.ts_Style = self.backup_style
        self.ts_Title = self.backup_title
        self.ts_Value = self.backup_value

        self.ts_Completion = False
        self.ts_PasswordMode = True
        self.ts_StripSpaces = True

##        (self.ts_Rect,
##         self.ts_ClientRect) = self.tsTextEntryDialogLayout(
##             parent, pos, size, style, caption)

        thePosition = self.Position
        theSize = self.Size

        if DEBUG:
            self.logger.debug('               self: %s' % self)
            self.logger.debug('             parent: %s' % parent)
            self.logger.debug('                 id: %s' % self.ts_Id)
            self.logger.debug('         AssignedId: %s' % self.ts_AssignedId)
            self.logger.debug('              title: %s' % title)
            self.logger.debug('                pos: %s' % thePosition)
            self.logger.debug('               size: %s' % theSize)
            self.logger.debug('              style: 0x%X' % style)
            self.logger.debug('               name: %s' % name)

        self.ts_Name = name
        self.ts_Parent = parent

        if True:
            self.ts_BackgroundColour = wx.ThemeToUse[
                'PasswordEntryDialog']['BackgroundColour'].lower()
            self.ts_ForegroundColour = wx.ThemeToUse[
                'PasswordEntryDialog']['ForegroundColour'].lower()
        else:
            self.ts_BackgroundColour = wx.ThemeToUse[
                'Dialog']['BackgroundColour'].lower()
            self.ts_ForegroundColour = wx.ThemeToUse[
                'Dialog']['ForegroundColour'].lower()

        self.ts_TextPad = None
        self.ts_IdleTimeQueue = wxDoubleLinkedList(lifoMode=False)
        self.ts_RealTimeQueue = wxDoubleLinkedList(lifoMode=False)

        self.ts_DescendantOrderOfShow = [self.ts_AssignedId]

        # The following redefines the usage of the style wx.CENTRE.
        # Intended only to govern the alignment of the buttons within
        # the dialog, we also apply it the centering of the dialog
        # itself within the desktop because the layout process cannot
        # wait for the application to initiate centering.
        if (self.ts_Style & wx.CENTRE) == wx.CENTRE:

            if DEBUG:

                msg = 'tsWxPasswordEntryDialog.__init__: ' + \
                      'Proceeding with tsDialogFeatureLayout. ' + \
                      'Style contains wx.CENTRE.'
                self.logger.debug(msg)
                print('DEBUG: %s\n' % msg)

            self.Center()

            # curses.noecho()

            self.ts_TextPad = self.tsDialogFeatureLayout()

            # curses.echo()

            self.Show()

        else:

            msg = 'tsWxPasswordEntryDialog.__init__: ' + \
                  'Skipped tsDialogFeatureLayout. ' + \
                  'Style did NOT contain wx.CENTRE.'
            self.logger.error(msg)
            print('ERROR: %s\n' % msg)

        self.ts_ButtonUserPressed = wx.ID_CANCEL

        self.tsEndClassRegistration(theClass)

#---------------------------------------------------------------------------

if __name__ == '__main__':

    print(__header__)
